### PR TITLE
feat(docs): Add aggregated metadata to website

### PR DESCRIPTION
**Pull request by joelspadin**
Added /hardware-metadata.json to the ZMK website. This gives external tools a way to grab all hardware metadata at once.

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -44,7 +44,9 @@
   },
   "devDependencies": {
     "@docusaurus/module-type-aliases": "^2.0.0-beta.3",
+    "@docusaurus/types": "^2.0.0-beta.18",
     "@tsconfig/docusaurus": "^1.0.2",
+    "@types/js-yaml": "^4.0.5",
```
Comment (joelspadin) [inline docs/package.json:49]: Pulled in types for js-yaml and docusaurus for type checking the plugin.

File: docs/package.json [hunk 2]
```diff
@@ -44,7 +44,9 @@
   },
   "devDependencies": {
     "@docusaurus/module-type-aliases": "^2.0.0-beta.3",
+    "@docusaurus/types": "^2.0.0-beta.18",
```
Comment (petejohanson) [inline docs/package.json:47]: Can we not pull in a version that matches our other docusaurus deps?

File: docs/package.json [hunk 3]
```diff
@@ -58,7 +60,7 @@
     "prebuild-webpack-plugin": "^1.1.1",
     "prettier": "2.3.1",
     "string-replace-loader": "^3.0.3",
-    "typescript": "^4.2.3",
+    "typescript": "^4.6.3",
```
[the remark this answers is not in the document]
Comment (petejohanson) [reply docs/package.json:63]: Still true if we stick w/ matching docusaurus dep versions?

**Review by petejohanson — CHANGES_REQUESTED** [review]
A few questions about the deps. Otherwise, looks good.

File: docs/package.json (hunk 2 above)
Comment (joelspadin) [reply docs/package.json]: Done.

**Comment by joelspadin** [comment]
For some reason I can't respond directly to your other comment. It looks like the older version of the types works with the older version of TypeScript.

**Review by petejohanson — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**